src/script/conversation/ConversationAccessPermission.ts

```diff
@@ -0,0 +1,145 @@
+/*
```

Comment:
Since we have some raw logic in there. I think it'd be very valuable to add unit tests to it.

Reply:
seems reasonable